src/commands/construct_transaction_command/transaction_actions/transfer_near_tokens_type/mod.rs

```diff
@@ -63,24 +63,38 @@ impl TransferNEARTokensAction {
                         }
                         None => crate::common::NearBalance::from_yoctonear(0),
                     };
+                let max_allowable_transfer_amount: crate::common::NearBalance =
+                    crate::common::get_max_allowable_transfer_amount(
+                        network_connection_config.clone(),
+                        sender_account_id.clone(),
+                    )?;
                 match item.amount {
                     Some(cli_amount) => {
-                        if cli_amount <= account_balance {
+                        if cli_amount <= max_allowable_transfer_amount {
                             cli_amount
                         } else {
                             println!(
                                 "You need to enter a value of no more than {}",
-                                account_balance
+                                max_allowable_transfer_amount
                             );
-                            TransferNEARTokensAction::input_amount(Some(account_balance))
+                            TransferNEARTokensAction::input_amount(
+                                Some(account_balance),
+                                max_allowable_transfer_amount,
+                            )
                         }
                     }
-                    None => TransferNEARTokensAction::input_amount(Some(account_balance)),
+                    None => TransferNEARTokensAction::input_amount(
+                        Some(account_balance),
+                        max_allowable_transfer_amount,
+                    ),
                 }
             }
             None => match item.amount {
                 Some(cli_amount) => cli_amount,
-                None => TransferNEARTokensAction::input_amount(None),
+                None => TransferNEARTokensAction::input_amount(
+                    None,
+                    crate::common::NearBalance::from_yoctonear(0),
+                ),
```

Comment:
As a rule of thumb, function name and its arguments should be such so it is easy to guess what the function will do. This function receives None and 0 balance - I have no good guesses based on this piece of information.

P.S. I believe the `input_amount` function does not need the account balance and only needs the transfer allowance as its upper limit.